### PR TITLE
Feature/cleanup

### PR DESCRIPTION
**Pull request by Robert-W**
Similar to route setter but simplifies configuration.

This PR removes the need to declare a profile inside two different spec properties and minimizes the boilerplate in this branch.

For example, when using this library, the configuration passed changed in the following manner:

```javascript
// Old
{
  profiles: {
    dstu2: {
      patient: {
        service: '...'
      }
    },
    stu3: {
      patient: {
        service: '...'
      }
    }
  }
}

// New
{
  profiles: {
    patient: {
      service: '...',
      versions: [ VERSIONS.DSTU2, VERSIONS.STU3 ]
    }
  }
}
```

We also changed routes in express from `dstu2/patient/_search` and `stu3/patient/_search` to `/:version/patient/_search`. This greatly reduces the amount of boilerplate necessary in this repo since we intend to support multiple versions going forward.

### CODE REVIEW

File: src/server/route-setter.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+				// Actual path for the route
+				route.path,
+				// Cors middleware
+				cors(cors_options),
+				// Version validation
+				versionValidationMiddleware(profile),
+				// Parameter sanitzation middleware
+				sanitizeMiddleware(route.args),
+				// Authentication middleware
+				validate(route.scopes, logger, config),
```
Comment (jonterrylee) [inline src/server/route-setter.js]: need to be able to turn off validate for capability statement

File: src/server/route-setter.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		// logger - application logger which get's injected into the routes
+		logger
+	} = options;
+
+	// Grab our user provided config
+	let { profiles, server } = config;
+
+	// Find and load all of our config files
+	let config_filepaths = glob.sync(path.join(process.cwd(), appConfig.files.configs));
+	let config_files = config_filepaths.map(filepath => require(filepath));
```
Comment (andrewmarcus) [inline src/server/route-setter.js]: Is there a change the `require` could fail?

File: src/server/profiles/observation/observation.config.js [hunk 3]
```diff
@@ -15,11 +16,14 @@ const scopes = [
 let routes = [
   {
     type: 'get',
-		path: '/dstu2/observation',
+		path: '/:version/observation',
```
Comment (andrewmarcus) [inline src/server/profiles/observation/observation.config.js]: Are there any differences between the DSTU2 and STU3 versions of the API endpoints?  Presumably the arguments are different, but as long as the endpoints themselves are the same I assume it is safe to combine them.

File: src/server/route-setter.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+		let { versions = [] } = profiles[profile_key];
+		versions.forEach(version => set.add(version));
+		return set;
+	}, new Set());
+
+	// Filter the provided versions by ones we actually support. We need to check this to make
+	// sure some user does not pass in a version we do not officially support in core for whatever
+	// reason. Otherwise there may be some compliance issues.
+	return Array.from(provided_versions)
+		.filter((version) => supported_versions.indexOf(version) !== -1);
```
Comment (andrewmarcus) [inline src/server/route-setter.js]: This is just a Set intersection, right?

**Pull request merged**